components/uuf-core/src/main/java/org/wso2/carbon/uuf/api/config/I18nResources.java

```diff
@@ -30,7 +30,7 @@
  */
 public class I18nResources {
 
-    private final Map<String, Properties> i18nResources = new HashMap<>();
+    private final SortedMap<String, Properties> i18nResources = new TreeMap<>();
```

Comment:
This is done to increase the performance. Keys in the tree maps will be sorted. So when we compare strings to get the i18n resource for a language (using startsWith in https://github.com/wso2/carbon-uuf/pull/159/files#diff-3aeccabd74d63e95c4ffca677a908412R58), we gain some performance improvement because we compare less elements.

As an example lets say we have following locales.
* en
* en-uk
* en-us

When we get **en** in the request, it matches with the **en** in the above list in the first try. So even though we use `startsWith` method, we get the **en** file because it is in the beginning of the list. If we get **en-uk**, it will be matched with the second element in the list.

But if we use a **HashMap**, the **key set** is not sorted, so we cannot say when it will be matched and we cannot sort the **keyset** for every request.